README.md

```diff
@@ -4,31 +4,95 @@ The `ray_julia_jll` is the Julia C++ wrapper that interfaces with the Ray.io pro
+
+Start by cloning this repo.  The following instructions assume you start from the repository root as the working directory:
+
+```sh
+git clone https://github.com/beacon-biosignals/Ray.jl.git
+cd Ray.jl
+```
+
+### Install dev dependnecies
+
+These are based on the dev setup instructions from upstream Ray but adapted here based on our fork.  These are the _minimal_ dependencies necessary to build the Ray server and the Ray.jl wrappers, and don't include e.g. the dependencies necessary for extras like the Ray dashboard.
+
+#### MacOS
+
+Based on [upstream Ray instructions](https://docs.ray.io/en/releases-2.5.1/ray-contribute/development.html#preparing-to-build-ray-on-macos):
+
+```sh
+brew update
+brew install wget
+brew install bazelisk
+```
+
+#### Linux
+
+Based on [upstream Ray instructions](https://docs.ray.io/en/releases-2.5.1/ray-contribute/development.html#preparing-to-build-ray-on-linux), assuming Ubuntu:
+
+```sh
+# Add a PPA containing gcc-9 for older versions of Ubuntu.
+sudo add-apt-repository -y ppa:ubuntu-toolchain-r/test
+sudo apt-get update
+sudo apt-get install -y build-essential curl gcc-9 g++-9 pkg-config psmisc unzip
+sudo update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-9 90 \
+              --slave /usr/bin/g++ g++ /usr/bin/g++-9 \
+              --slave /usr/bin/gcov gcov /usr/bin/gcov-9
+```
+
+Install [Bazelisk](https://github.com/bazelbuild/bazelisk#readme) by downloading the latest binary for your system from the [releases page](https://github.com/bazelbuild/bazelisk/releases) somewhere on your `$PATH` (like `$HOME/bin/`).  For instance, as of writing this was the latest release for an AMD64 system:
+
+```sh
+curl -Lo "$HOME/bin/bazel" "https://github.com/bazelbuild/bazelisk/releases/download/v1.18.0/bazelisk-linux-amd64"
+chmod +x "$HOME/bin/bazel"
+```
+
+Alternatively, if you have `npm` installed you can install Bazelisk that way:
+
+```sh
+npm install -g @bazel/bazelisk
+```
+
+### Prepare python environment
+
+We recommend always using the same virtualenv as Bazel can cause unnecessary rebuilds when using different versions of Python.
+
+```sh
+python -m venv venv
+source venv/bin/activate
+# make sure we're using up-to-date version of pip and wheel:
+python -m pip install --upgrade pip wheel
+```
+
+(These are the same as the [upstream Ray instructions](https://docs.ray.io/en/releases-2.5.1/ray-contribute/development.html#prepare-the-python-environment) but copied here for consistency.)
+
+### Build wrapper
 
 ```sh
-# We recommend always using the same virtualenv as Bazel can cause unnecessary rebuilds when
-# using different versions of Python.
 python3 -m venv venv
 source venv/bin/activate
 
 # Build the required libraries
 julia --project=ray_julia_jll -e 'using Pkg; Pkg.build(verbose=true)'
+```
+
+### Build Ray CLI/server
 
```

Comment:
```suggestion
From within the same virtual environment you can also easily install the ray cli by running the command below. First, modify the url below to correspond to the version of python you are using (in this instance it's `python3.8`).

```sh
pip install https://github.com/beacon-biosignals/ray/releases/download/ray-2.5.1-beacon/ray-2.5.1-cp38-cp38-manylinux2014_x86_64.whl
```

On Mac it's:
```sh
pip install https://github.com/beacon-biosignals/ray/releases/download/ray-2.5.1-beacon/ray-2.5.1-cp38-cp38-macosx_13_0_arm64.whl
```

Alternatively, to build ray from source (which takes ~30 mins) you can follow these steps:
```

Reply:
the suggestion is getting messed up by the ```suggestion` block but you get the idea

Reply:
I thought I had these but maybe we were racin'